ee/src/main/java/org/jboss/as/ee/subsystem/GlobalDirectoryResourceDefinition.java

```diff
@@ -0,0 +1,175 @@
+
+        public GlobalDirectoryAddHandler() {
+            super(ATTRIBUTES);
+        }
+
+        @Override
+        protected void populateModel(final OperationContext context, final ModelNode operation, final Resource resource)
+                throws OperationFailedException {
+            super.populateModel(context, operation, resource);
+            context.addStep(new OperationStepHandler(){
+                public void execute(OperationContext oc, ModelNode op) throws OperationFailedException {
+                    Resource parentResource = context.readResourceFromRoot(context.getCurrentAddress().getParent(), false);
+                    Set<String> globalDirectories = parentResource.getChildrenNames(EESubsystemModel.GLOBAL_DIRECTORY);
+
+                    String newName = context.getCurrentAddressValue();
+                    String firstAdded = !globalDirectories.isEmpty() ? globalDirectories.iterator().next() : null;
+                    if (firstAdded != null && !firstAdded.equals(newName)) {
+                        throw EeLogger.ROOT_LOGGER.oneGlobalDirectory(newName, firstAdded);
+                    }
+                }
+            }, MODEL);
+        }
+
+        @Override
+        protected void performRuntime(OperationContext context, ModelNode operation, ModelNode model) throws OperationFailedException {
+            final String path = PATH.resolveModelAttribute(context, model).asString();
+            final String relativeTo = RELATIVE_TO.resolveModelAttribute(context, model).asStringOrNull();
+
+            final CapabilityServiceBuilder<?> serviceBuilder = context.getCapabilityServiceTarget()
+                    .addCapability(EE_GLOBAL_DIRECTORY_CAPABILITY);
+            final Consumer<GlobalDirectory> provides = serviceBuilder.provides(EE_GLOBAL_DIRECTORY_CAPABILITY);
+            final Supplier<PathManager> pathManagerSupplier = serviceBuilder.requiresCapability(PATH_MANAGER_CAPABILITY, PathManager.class);
+
+            Service globalDirectoryService = new GlobalDirectoryService(pathManagerSupplier, provides, context.getCurrentAddressValue(), path, relativeTo);
+
+            serviceBuilder.setInstance(globalDirectoryService)
+                    .setInitialMode(ServiceController.Mode.ACTIVE)
+                    .install();
+        }
+    }
+
+    @Override
+    public void registerAttributes(ManagementResourceRegistration resourceRegistration) {
+        ReloadRequiredWriteAttributeHandler handler = new ReloadRequiredWriteAttributeHandler(ATTRIBUTES);
+        for (AttributeDefinition attribute : ATTRIBUTES) {
+            resourceRegistration.registerReadWriteAttribute(attribute, null, handler);
+        }
+    }
+
+    private static class GlobalDirectoryRemoveHandler extends AbstractRemoveStepHandler {
+
+        @Override
+        protected void performRuntime(OperationContext context, ModelNode operation, ModelNode model) throws OperationFailedException {
+            context.restartRequired();
```

Comment:
This seems odd. Other dynamically created modules, e.g. deployments, don't require a full process restart. I'd ask about .setRemoveRestartLevel(OperationEntry.Flag.RESTART_JVM) above as well but I'll just ask here.

Reply:
Fixed, now we are using a ReloadRequiredRemoveStepHandler